libkbfs/block_retrieval_queue.go

```diff
@@ -174,35 +174,38 @@ func (brq *blockRetrievalQueue) notifyWorker() {
 func (brq *blockRetrievalQueue) CacheAndPrefetch(ctx context.Context,
 	ptr BlockPointer, block Block, kmd KeyMetadata, priority int,
 	lifetime BlockCacheLifetime, hasPrefetched bool) (err error) {
-	dbc := brq.config.DiskBlockCache()
-	if dbc != nil {
-		go func() {
-			if err := dbc.UpdateLRUTime(ctx, ptr.ID); err != nil {
-				brq.log.CWarningf(ctx, "Error updating metadata: %+v", err)
-			}
-		}()
-	}
 	defer func() {
 		if err != nil {
 			brq.log.CWarningf(ctx, "Error Putting into the block cache: %+v",
 				err)
 		}
+		dbc := brq.config.DiskBlockCache()
+		if dbc != nil {
+			go func() {
+				err := dbc.UpdateMetadata(ctx, ptr.ID, hasPrefetched)
+				if err != nil {
+					brq.log.CWarningf(ctx, "Error updating metadata: %+v", err)
```

Comment:
Can we take this opportunity to avoid printing a warning in the "block not found" case? Seems pretty common when I look through logs.